### ISSUE TITLE
issue#9 - Implementing Variable Pauses After I2C Register Writes in i2c_ctrl Module

### ISSUE THREAD

**Issue by Juninho99** [chili-chips-ba/openeye-CamSI#29]
## Description:
According to the camera's specification (**Sony IMX283**), after certain registers are written, there needs to be a pause of a few milliseconds before proceeding to the next set of registers.

## Objective:
Modify the existing `i2c_ctrl` module and `i2c_top` module to implement variable pauses after specific I2C register writes, as specified by the camera's requirements.

## Problem Statement:
The challenge is to introduce variable pauses after writing certain registers, as required by the camera's specification. These pauses are not periodic and occur only after specific registers. The pauses need to be controlled dynamically, allowing for different durations specified in milliseconds.

## Proposed Solution:
- Modify the `i2c_init.mem` file to include the pause duration for each register write.
- Update the `i2c_data_init` array in the `i2c_top` module to read the pause durations along with register addresses and data.
- Adjust the `i2c_ctrl` module to implement a **WAIT** state that counts the required pause duration using nested counters, avoiding resource-intensive multiplication.
- Ensure that pauses are dynamically controlled based on the data read from i2c_init.mem.

**Comment by chili-chips-ba** [comment]
Differences between 2-lane and 4-lane cameras should not be hard coded in the main RTL body. Instead, they should go into customization package under corresponding branches, based on the single master selector.

![image](https://github.com/user-attachments/assets/64ef5226-e619-4860-b0a6-09fc49d5f299)
-----
![image](https://github.com/user-attachments/assets/284f89b7-f2e7-4f98-b143-d083c5a2b76d)

**Comment by Juninho99** [comment]
You are right. I will make some changes and push them on issue#9 branch. Also I'll make pull request asap.

**Comment by Juninho99** [comment]
The I2C part is fully parameterized and successfully tested on 4-lane sensor. However, we will keep this issue open until we have tested the I2C on both 2-lane sensors.

**Comment by Juninho99** [comment]
IMX219 is also tested successfully. There is still a webcam left.

**Comment by chili-chips-ba** [comment]
Hi @Juninho99 what is the latest status on this track? Can we now close it?
